### PR TITLE
Add matrix release alert script and gitlab job

### PR DESCRIPTION
**Pull request by s3krit**
This script should run whenever a tag formatted like 'v1.2.3' is detected, and will send an alert to Matrix that there is a pending release - provided the tag is signed by an approved GPG signing key.

### CODE REVIEW

**Comment by lovelaced** [comment]
Looks reasonable - could you add a a commented Usage: section at the top of the script explaining what arguments are expected and how many?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -111,8 +111,14 @@ check-line-width:
...
   allow_failure:                   true
 
-
-
+alert-pending-release:
+  stage:                           test
+  image:                           parity/tools:latest
+  only:
+    - tags
+    - /^v[0-9]+\.[0-9]+\.[0-9].*$/        # i.e. v1.0.1, v2.1.0rc1
```
Comment (kirushik) [inline .gitlab-ci.yml]:
```suggestion
    - /^v[0-9]+\.[0-9]+\.[0-9]+.*$/        # i.e. v1.0.1, v2.1.0rc1
```

I guess `v0.1.11` is a pretty valid tag =)

Comment (kirushik) [reply .gitlab-ci.yml]: Hmm, actually the original regexp would of course work — it just feels more natural with all digits being captured logically

File: scripts/gitlab/alert_pending_release.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
+#!/bin/bash
+
+# structure_message $content $formatted_content (optional)
+structure_message() {
+  if [ -z "$2" ]; then
+    body=$(jq -Rs --arg body "$1" '{"msgtype": "m.text", $body}' < /dev/null)
```
Comment (kirushik) [inline scripts/gitlab/alert_pending_release.sh]: Does `parity/tools` have `jq` installed?

Comment (s3krit) [reply scripts/gitlab/alert_pending_release.sh]: Yep.

**Pull request merged**